src/dfx/src/lib/operations/canister/deploy_canisters.rs

```diff
@@ -193,16 +194,18 @@ async fn register_canisters(
 #[context("Failed to build call canisters.")]
 async fn build_canisters(
     env: &dyn Environment,
-    canister_names: &[String],
+    referenced_canisters: &[String],
```

Comment:
It looks like you're taking care of this here, but I wanted to mention: one of the things to make sure of here is that referenced canister ids are still passed to moc, and in environment variables, for remote canisters.

Reply:
Yes, this is what I do to still supply the ids. This is tested for moc in `remote.bash` in test `build + install + call -- remote`. I also added a test that makes sure the environment variables are set correctly, too.